src/mpi/coll/iallgather/iallgather_tsp_brucks_algos.h

```diff
@@ -67,12 +67,14 @@ MPIR_TSP_Iallgather_sched_intra_brucks(const void *sendbuf, int sendcount,
     /* get datatype info of sendtype and recvtype */
     MPIR_Datatype_get_size_macro(sendtype, sendtype_size);
     MPIR_Datatype_get_extent_macro(sendtype, sendtype_extent);
-    MPIR_Type_get_true_extent_impl(sendtype, &sendtype_lb, &sendtype_true_extent);
+    MPIR_Type_get_true_extent_impl(sendtype, (MPI_Aint *) & sendtype_lb,
```

Comment:
These additional casts (which is presume are squashing warnings?) don't belong with this argument re-ordering, IMO.